lib/semantic-toast.jsx

```diff
@@ -7,38 +7,48 @@ import 'semantic-ui-css/components/header.min.css';
 
 import withTransition from './with-transition';
 
-const icons = {
-    info: 'announcement',
-    success: 'checkmark',
-    error: 'remove',
-    warning: 'warning circle'
-};
-
-function SemanticToast(props) {
-    const { type, title, description, onClose } = props;
-    const icon = props.icon || icons[props.type];
-
-    return (
-        <Message
-            {...{ [type]: true }}
-            onDismiss={onClose}
-            header={title}
-            content={description}
-            icon={icon}
-            floating
-        />
-    );
+class SemanticToast extends PureComponent {
```

Comment:
No need to turn `SemanticToast` into a class component, there is no state to store here. All you have to do is to define your `onDismiss` function inside the functional stateless component, just before the `return`; 

This way, yo will have access to the `onClose` prop, which is on scope:

```
function SemanticToast(props) {
    const { type, title, description, onClick, onClose } = props;
    const icon = props.icon || icons[props.type];

    const onDismiss = e => {
        e.stopPropagation();
        onClose();
    }

    return (
        <Message
            {...{ [type]: true }}
            onDismiss={onDismiss}
            header={title}
            content={description}
            icon={icon}
            floating
        />
    );
}
```

Reply:
Good. I thought declaring a function inside a function is considered bad tone.

Reply:
Its not bad per itself, it's just that the inner function is created every single time the outer function is called, which now that I think about it, its the exact same thing that would happen on your first implementation :)

So we do not have that benefit, but its not important here and it boils down to a question of style.. I rather have it this way than turning it into a class component with no state.

If we would re-render this component many times and performance would be an issue, than I would turn it into a class component to avoid re-creating the `onDismiss` function on every render.